src/EffectiveCSharp.Analyzers/PreferDeclarationInitializersToAssignmentStatementsAnalyzer.cs

```diff
@@ -0,0 +1,637 @@
+namespace EffectiveCSharp.Analyzers;
+
+/// <summary>
+/// A <see cref="DiagnosticAnalyzer" /> for Effective C# Item #12 - Prefer field declaration initializers.
+/// </summary>
+/// <seealso cref="DiagnosticAnalyzer" />
+[DiagnosticAnalyzer(LanguageNames.CSharp)]
+public class PreferDeclarationInitializersToAssignmentStatementsAnalyzer : DiagnosticAnalyzer
+{
+    private static readonly string HelpLinkUri = $"https://github.com/rjmurillo/EffectiveCSharp.Analyzers/blob/{ThisAssembly.GitCommitId}/docs/rules/{DiagnosticIds.PreferDeclarationInitializersToAssignmentStatement}.md";
+
+    private static readonly DiagnosticDescriptor GeneralRule = new(
+        id: DiagnosticIds.PreferDeclarationInitializersToAssignmentStatement,
+        title: "Prefer field declaration initializers to assignment statements",
+        messageFormat: "Use a field declaration initializer instead of an assignment statement",
+        category: "Maintainability",
+        defaultSeverity: DiagnosticSeverity.Info,
```

Comment:
We want to use Info in order to reduce the likelihood that users turn off the analyzers since it is possible legacy code bases will trigger a lot of diagnostics.